storage_drivers/azure/api/azure.go

```diff
@@ -129,17 +134,42 @@ type Client struct {
 }
 
 // NewDriver is a factory method for creating a new SDK interface.
-func NewDriver(config ClientConfig) (Azure, error) {
+func NewDriver(ctx context.Context, config ClientConfig) (Azure, error) {
 	var err error
+	var credential azcore.TokenCredential
 
 	// Ensure we got a location
 	if config.Location == "" {
 		return nil, errors.New("location must be specified in the config")
 	}
 
-	credential, err := azidentity.NewClientSecretCredential(config.TenantID, config.ClientID, config.ClientSecret, nil)
-	if err != nil {
-		return nil, err
+	if config.ClientSecret == "" && config.ClientID == "" {
```

Comment:
I think the logic is covered in https://github.com/kubernetes-sigs/cloud-provider-azure/blob/e1e1b0f37fea5383db5191595c9327cbc35c44a5/pkg/azclient/auth.go#L120

Reply:
This `clientSecret` and `clientID` are from an external configuration if user configured explicitly, not from azure.json. If users rely on azure.json for authentication, we uses `authProvider.GetAzIdentity()` to get credential.

Reply:
clientid client secret can be assigned to authconfig struct and then we can create auth provider using this config...

Reply:
That's right, I've made changes to the PR.